scrapers_next/mt/committees.py

```diff
@@ -0,0 +1,185 @@
+        house_prefix = "(H) "
+        senate_prefix = "(S) "
+        if committee_title.startswith(house_prefix):
+            chamber = "lower"
+            committee_title = committee_title[len(house_prefix) :]
+        if committee_title.startswith(senate_prefix):
+            chamber = "upper"
+            committee_title = committee_title[len(senate_prefix) :]
+
+        # 2) A joint prefix may exist
+        joint_prefix = "Joint "
+        if committee_title.startswith(joint_prefix):
+            chamber = "legislature"
+            committee_title = committee_title[len(joint_prefix) :]
+
+        # 3) Select committee text isn't part of the name
+        committee_title = committee_title.replace("Select Committee on ", "")
+
+        committee_suffix = " Committee"
+        if committee_title.endswith(committee_suffix):
+            committee_title = committee_title[: -len(committee_suffix)]
+
+        # 4) Check if a subcommittee
+        subcommittee_info = committee_title.split(" Subcommittee on ")
+        if len(subcommittee_info) == 2:
+            committee_title = subcommittee_info[1]
+            classification = "subcommittee"
+            parent = subcommittee_info[0]
+
+        # Build the committee object with calculated metadata
+        com = ScrapeCommittee(
+            name=committee_title,
+            chamber=chamber,
+            classification=classification,
+            parent=parent,
+        )
+        com.add_source(member_info_href, note="Committee membership page")
+        com.add_link(member_info_href, note="homepage")
+
+        return CommitteeMembersPage(
+            {"com": com}, source=URL(member_info_href, timeout=30)
+        )
+
+
+class CommitteeMembersPage(HtmlPage):
+    def process_page(self):
+        com = self.input.get("com")
+        # The member table is the last table
+        table = XPath("//table").match(self.root)[-1]
+        rows = XPath("./tr[position()>1]").match(table)
+        for row in rows:
+            # Member name is inside of an <a> tag
+            try:
+                member = XPath("td[1]/a/text()").match_one(row)
+            except SelectorError:
+                # If member name isn't in an <a> tag, they are staff so skip
+                continue
+
+            # Member names may have double spaces, replace with single spaces
+            member = member.replace("  ", " ")
+
+            # Role is listed in the second column
+            role = XPath("td[2]/text()").match_one(row)
+
+            com.add_member(name=member, role=role)
+
+        # Skip if no members found
+        if len(com.members) == 0:
+            raise SkipItem(f"No membership data found for: {com.name}")
+
+        return com
+
+
+# This page contains information on the Administrative committees only
+# This site is a little buggy or perhaps there's a rate limiter involved
+# Sometimes requests to this site do not return the expected html
+class AdministrativeCommitteeList(HtmlListPage):
```

Comment:
We want to get Interim Committees too (let's reformat the code to be able to do that even though the actual committees haven't been fully updated with new session membership yet). Good news is, I tested it out, and your existing code for handing Administrative Committees will also work for Interim.

You'll just need to:
- Rename this class `OtherCommList` and remove the `source` attribute defined within it.
- Define the class instances of `Administrative(OtherCommList)` and `Interim(OtherCommList)` as child classes with their respective source attributes defined within each child class (for interim it'll be https://leg.mt.gov/committees/interim/)
- Then, in the details class, add an additional selector that grabs the alternative name of the committee using xpath `".//div[@class='container white']/h2"`, which, in cases like the [2021-2022 Children, Families, Health, and Human Services Interim Committee](https://leg.mt.gov/committees/interim/cfhhs/) gets the more complete name than the string `"2021-2022 CFHHS"` which is what the current selector grabs
   - Then, compare the length of the two strings, and if the one you're currently grabbing from the breadcrumb items is shorter, concatenate it to the long one (after any date range is removed, perhaps using regex).
   - So, for this example, the final saved committee name should be `"Children, Families, Health, and Human Services Interim (CFHHS)"
- **Note: just as you should allow the word "Legislative" to stay in the name, also keep "Interim" to help distinguish from any standing committees that may have the same name as interim committees (important for us internally when we're matching / referencing committee names in bills).**

- *Also, add a conditional check in the list page class, that checks if "interim" is in `self.source.url`. If it is, use `logging.warning` to log a warning to the console something like: "Interim Committee data may not be updated yet for this current session: a manual check should be conducted". (of course, you'll need to import `logging`)*

Let me know if you have any questions while executing on this change